es-app/src/views/SystemView.cpp

```diff
@@ -278,9 +287,27 @@ void SystemView::render(const Eigen::Affine3f& parentTrans)
 	
 	Eigen::Affine3f trans = getTransform() * parentTrans;
 
-	renderExtras(trans);
-	renderCarousel(trans);
-	renderInfoBar(trans);
+	auto systemInfoZIndex = mSystemInfo.getZIndex();
+	auto minMax = std::minmax(mCarousel.zIndex, systemInfoZIndex);
+
+	renderExtras(trans, INT16_MIN, minMax.first);
+	renderFade(trans);
+
+	if (mCarousel.zIndex > mSystemInfo.getZIndex()) {
+		renderInfoBar(trans);
+	} else {
+		renderCarousel(trans);
+	}
+
+	renderExtras(trans, minMax.first, minMax.second);
+
+	if (mCarousel.zIndex > mSystemInfo.getZIndex()) {
+		renderCarousel(trans);
+	} else {
+		renderInfoBar(trans);
+	}
```

Comment:
I agree.  It is further complicated by the fact that we have to render multiple sets of extras in the proper z-index order to ensure the slide transition works properly.